vowpalwabbit/core/include/vw/core/multi_model_utils.h

```diff
@@ -7,49 +7,92 @@
 #include "vw/core/array_parameters_dense.h"
 #include "vw/core/learner.h"
 
-#include <queue>
-
 namespace VW
 {
 namespace reductions
 {
 namespace multi_model
 {
-// ***** NOTE: params_per_problem must be of form 2^n *****
-inline void clear_offset(dense_parameters& weights, const size_t offset, const size_t params_per_problem)
+// ***** NOTE: ppw must be of form 2^n *****
+inline void clear_offset(dense_parameters& weights, const size_t offset, const size_t ppw, const size_t inner_ppw_size)
```

Comment:
this technically should be recursive, and maybe even part of learner.h. in this impl we can only have two reductions that modify the wpp, if we add a third it would stop working, right?